dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python2
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+import os
+import sys
+import json
+import subprocess
+
+from sparktestsupport import SPARK_HOME, ERROR_CODES
+from sparktestsupport.shellutils import exit_from_command_with_retcode, run_cmd, rm_r
+
+
+def print_err(*args):
+    """
+    Given a set of arguments, will print them to the STDERR stream
+    """
+    print(*args, file=sys.stderr)
+
+
+def post_message(mssg, comments_url):
+    http_code_header = "HTTP Response Code: "
+    posted_message = json.dumps({"body": mssg})
+
+    print("Attempting to post to Github...")
+
+    # we don't want to call `run_cmd` here as, in the event of an error, we DO NOT
+    # want to print the GITHUB_OAUTH_KEY into the public Jenkins logs
+    curl_proc = subprocess.Popen(['curl',
```

Comment:
Same comment applies here RE: indentation; it might be better to indent these four spaces or to split the arguments into a variable definition on a previous line.